.gitignore

```diff
@@ -196,10 +196,6 @@ Temporary Items
 debug/
 target/
 
-# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
-# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
-Cargo.lock
-
```

Comment:
Not against tracking the Cargo.lock, but still wondering: do you have a specific reason for adding it?

Reply:
Yes. Without the Cargo.lock file this build isn't reproducible. Nix is all about reproducible builds. Cargo.lock should generally be included for applications and not for libraries.

From the link referenced here:
> If you’re building a non-end product, such as a rust library that other rust [packages](https://doc.rust-lang.org/cargo/appendix/glossary.html#package) will depend on, put Cargo.lock in your .gitignore. If you’re building an end product, which are executable like command-line tool or an application, or a system library with crate-type of staticlib or cdylib, check Cargo.lock into git.

we're building an application here.

Reply:
> Cargo.lock should generally be included for applications and not for libraries.

:+1: